File: javascript/7_kyu/letterbox_paint_squad.js
// Letterbox Paint-Squad
// https://www.codewars.com/kata/597d75744f4190857a00008d/train/javascript

var paintLetterboxes = function(start, end) {
    var arr = [];
    for (var i = start; i < end + 1; i++) {
        arr.push(String(i).split(''));
    }

    numbers = {};

    for (var i = 0; i < arr.length; i++) {
        for (var j = 0; j < arr[i].length; j++) {
            if ( arr[i][j] in numbers ) {
                numbers[arr[i][j]]++;
            } else {
                numbers[arr[i][j]] = 1;
            }
        }
    }

    var ans = [];

    for (var i = 0; i < 10; i++) {
        if ( String(i) in numbers) {
            ans.push(numbers[String(i)]);
        } else {
            ans.push(0);
        }
    }

    return ans;
}

console.log(paintLetterboxes(125, 132))

File: python/7_kyu/letterbox_paint_squad.py
# Letterbox Paint-Squad
# https://www.codewars.com/kata/597d75744f4190857a00008d/train/python

def paint_letterboxes(start, finish):
    number_dic = {}
    arr = []

    for i in range(start, finish + 1):
        splitting = list(str(i))
        for j in splitting:
            arr.append(j)

    for i in arr:
        if int(i) in number_dic:
            number_dic[int(i)] += 1
        else:
            number_dic[int(i)] = 1

    ans = []

    for i in range(0, 10):
        if i in number_dic:
            ans.append(number_dic[i])
        else:
            ans.append(0)

    return ans




print(paint_letterboxes(125, 132))